read_input.l

```diff
@@ -1504,6 +1504,35 @@ static inline double fltlist_next_token(int * const list_end){
       }
     }
   }
+  {SPC}*MGCoarseSolverType{EQL}{STRLIST} {
+    char paramname[100];
+    char token[100];
+    char error_message[200];
+    int list_end = 0;
+    int level = 0;
+    strlist_tokenize(yytext, paramname, 100);
+    strlist_next_token(&list_end, token, 100);
+    while( list_end != 1 ){
+      if( level > QUDA_MAX_MG_LEVEL ){
+        snprintf(error_message, 200, "Exceeded maximum number of levels (%d) parsing %s!\n", QUDA_MAX_MG_LEVEL, paramname);
+        yy_fatal_error(error_message);
+
+      }
+      if( strcmp(token, "cagcr") == 0 ){
+        quda_input.mg_coarse_solver_type[level]=QUDA_CA_GCR_INVERTER;
+        level++;
```

Comment:
```level++``` should be in the outer scope (one line instead of 3)
There should be an error condition for invalid input here as well as in the smoother decision tree